src/jobs/publish_s3.yml

```diff
@@ -0,0 +1,35 @@
+description: Publish files to S3
+
+executor: aws
+
+parameters:
+  public:
+    type: boolean
+    description: Whether to store in the public bucket
+  path:
```

Comment:
If possible I would like that this orb would "abstract" some of this things and gives less power to the user.

What I mean is, we can maybe pass the "repository-name" and "version", or something like that, and then the path would be correctly built here, without the need of each component knowing if we usually do a "codacy/bucket", or "codacy/internal", and stuff like that. What are your thoughts about it?

Reply:
What path do you suggest for a file?
Something like `https://artifacts.codacy.com/codacy-coverage-reporter/binaries/1.0.0/codacy-coverage-reporter-linux` ?
To do this we could ask the component only the `path: binaries` and evething else is generated from the version and the repository name.

Note: Changing the repository name changes the paths.

Reply:
path-prefix / category  or something that show the intent of what we are publishing. to be a mandatory non-empty string (assuming that we cannot have enums here). And we can set it to either api or binaries.

Or we can just skip all of this binaries / api, since maybe it is not that worth? and we just focus on:

`https://artifacts.codacy.com/<public>/codacy-coverage-reporter/1.0.0/codacy-coverage-reporter-linux`

Either way, would not add the "binaries" between the name of the component and the 1.0.0, since that "path" should be more like a "bucket name", not something in the middle, imo